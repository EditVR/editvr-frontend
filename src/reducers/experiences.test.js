/**
 * @file experiences.test.js
 * Contains tests for experiences.js.
 */

import reducer from './experiences';
import {
  EXPERIENCES_FETCH_FOR_USER,
  EXPERIENCES_CREATE,
  EXPERIENCES_EDIT
} from '../constants';

describe('reducers->experiences', () => {
  it('Should return the initial state by default', () => {
    expect(reducer(undefined, {})).toMatchSnapshot();
  });

  it(`Should handle ${EXPERIENCES_FETCH_FOR_USER}_LOADING`, () => {
    expect(
      reducer(undefined, {
        type: `${EXPERIENCES_FETCH_FOR_USER}_LOADING`,
        loading: true,
        error: null,
        payload: {}
      })
    ).toMatchSnapshot();
  });

  it(`Should handle ${EXPERIENCES_FETCH_FOR_USER}_FAIL`, () => {
    const error = 'Error: failed to fetch experiences.';
    expect(
      reducer(undefined, {
        type: `${EXPERIENCES_FETCH_FOR_USER}_FAIL`,
        loading: false,
        payload: { error }
      })
    ).toMatchSnapshot();
  });

  it(`Should handle ${EXPERIENCES_FETCH_FOR_USER}_SUCCESS`, () => {
    expect(
      reducer(undefined, {
        type: `${EXPERIENCES_FETCH_FOR_USER}_SUCCESS`,
        loading: false,
        error: null,
        payload: [
          {
            id: 1,
            title: 'example experience'
          }
        ]
      })
    ).toMatchSnapshot();
  });

  it(`Should handle ${EXPERIENCES_CREATE}_LOADING`, () => {
    expect(
      reducer(undefined, {
        type: `${EXPERIENCES_CREATE}_LOADING`,
        loading: true,
        error: null,
        payload: {}
      })
    ).toMatchSnapshot();
  });

  it(`Should handle ${EXPERIENCES_CREATE}_FAIL`, () => {
    const error = 'Error: failed to fetch experiences.';
    expect(
      reducer(undefined, {
        type: `${EXPERIENCES_CREATE}_FAIL`,
        loading: false,
        payload: {
          error
        }
      })
    ).toMatchSnapshot();
  });

  it(`Should handle ${EXPERIENCES_CREATE}_SUCCESS`, () => {
    expect(
      reducer(undefined, {
        type: `${EXPERIENCES_CREATE}_SUCCESS`,
        loading: false,
        error: null,
        payload: [
          {
            id: 1,
            title: 'example experience'
          }
        ]
      })
    ).toMatchSnapshot();
  });

  it(`Should handle ${EXPERIENCES_EDIT}_LOADING`, () => {
    expect(
      reducer(undefined, {
        type: `${EXPERIENCES_EDIT}_LOADING`,
        loading: true,
        error: null,
        payload: {}
      })
    ).toMatchSnapshot();
  });

  it(`Should handle ${EXPERIENCES_EDIT}_FAIL`, () => {
    const error = 'Error: failed to fetch experiences.';
    expect(
      reducer(undefined, {
        type: `${EXPERIENCES_EDIT}_FAIL`,
        loading: false,
        payload: { error }
      })
    ).toMatchSnapshot();
  });

  it(`Should handle ${EXPERIENCES_EDIT}_SUCCESS`, () => {
    // Pass in mock state, this will assert that this reducer is injecting
    // the updated experience at the right index.
    const state = {
      items: [
        {
          id: 0
        },
        {
          id: 3
        },
        {
          id: 1
        },
        {
          id: 2
        }
      ]
    };

    expect(
      reducer(state, {
        type: `${EXPERIENCES_EDIT}_SUCCESS`,
        loading: false,
        error: null,
        payload: {
          id: 1,
          title: 'example experience'
        }
      })
    ).toMatchSnapshot();
  });
});
