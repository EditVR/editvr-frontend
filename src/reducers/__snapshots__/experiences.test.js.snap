// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers->experiences Should handle EXPERIENCES_CREATE_FAIL 1`] = `
Object {
  "error": "Error: failed to fetch experiences.",
  "items": Array [],
  "loading": false,
}
`;

exports[`reducers->experiences Should handle EXPERIENCES_CREATE_LOADING 1`] = `
Object {
  "error": null,
  "items": Array [],
  "loading": true,
}
`;

exports[`reducers->experiences Should handle EXPERIENCES_CREATE_SUCCESS 1`] = `
Object {
  "error": null,
  "items": Array [
    Array [
      Object {
        "id": 1,
        "title": "example experience",
      },
    ],
  ],
  "loading": false,
}
`;

exports[`reducers->experiences Should handle EXPERIENCES_EDIT_FAIL 1`] = `
Object {
  "error": "Error: failed to fetch experiences.",
  "items": Array [],
  "loading": false,
}
`;

exports[`reducers->experiences Should handle EXPERIENCES_EDIT_LOADING 1`] = `
Object {
  "error": null,
  "items": Array [],
  "loading": true,
}
`;

exports[`reducers->experiences Should handle EXPERIENCES_EDIT_SUCCESS 1`] = `
Object {
  "error": null,
  "items": Array [
    Object {
      "id": 0,
    },
    Object {
      "id": 3,
    },
    Object {
      "id": 1,
      "title": "example experience",
    },
    Object {
      "id": 2,
    },
  ],
  "loading": false,
}
`;

exports[`reducers->experiences Should handle EXPERIENCES_FETCH_FOR_USER_FAIL 1`] = `
Object {
  "error": "Error: failed to fetch experiences.",
  "items": Array [],
  "loading": false,
}
`;

exports[`reducers->experiences Should handle EXPERIENCES_FETCH_FOR_USER_LOADING 1`] = `
Object {
  "error": null,
  "items": Array [],
  "loading": true,
}
`;

exports[`reducers->experiences Should handle EXPERIENCES_FETCH_FOR_USER_SUCCESS 1`] = `
Object {
  "error": null,
  "items": Array [
    Object {
      "id": 1,
      "title": "example experience",
    },
  ],
  "loading": false,
}
`;

exports[`reducers->experiences Should return the initial state by default 1`] = `
Object {
  "error": null,
  "items": Array [],
  "loading": false,
}
`;
